index.bs

```diff
@@ -209,83 +249,83 @@ enum OperandType {
   "uint8"
 };
 
-dictionary OperandDescriptor {
+dictionary MLOperandDescriptor {
   // The operand type.
-  required OperandType type;
+  required MLOperandType type;
 
   // The dimensions field is only required for tensor operands.
   // The negative value means an unknown dimension.
   sequence<long> dimensions;
 };
 </script>
 
-## Operand ## {#api-operand}
+## MLOperand ## {#api-mloperand}
 <script type=idl>
 [SecureContext, Exposed=Window]
-interface Operand {};
+interface MLOperand {};
 </script>
 
-## NeuralNetworkContext ## {#api-neuralnetworkcontext}
+## MLModelBuilder ## {#api-mlmodelbuilder}
 
-The {{NeuralNetworkContext}} interface represents a global state of neural network compute workload and execution processes.
+The {{MLModelBuilder}} interface defines a set of operations as identified by the [[#usecases]] that can be composed into a computational graph. It also represents the intermediate state of a graph building session.
 
 <script type=idl>
-[SecureContext, Exposed=Window]
-interface NeuralNetworkContext {
-  ModelBuilder createModelBuilder();
-};
-</script>
+typedef record<DOMString, MLOperand> MLNamedOperands;
 
-## ModelBuilder ## {#api-modelbuilder}
+[SecureContext, Exposed=Window]
+interface MLModelBuilder {
+  // Construct the model builder from the context.
+  constructor(MLContext context);
 
-The {{ModelBuilder}} interface defines a set of operations as identified by the [[#usecases]] that can be composed into a computational graph. It also represents the intermediate state of a graph building session.
+  // Create an operand for a model input.
+  MLOperand input(DOMString name, MLOperandDescriptor desc);
 
-<script type=idl>
-typedef record<DOMString, Operand> NamedOperands;
+  // Create an operand for a model constant.
+  MLOperand constant(MLOperandDescriptor desc, ArrayBufferView bufferView);
 
-[SecureContext, Exposed=Window]
-interface ModelBuilder {
-  // Create an operand that represents a model input.
-  Operand input(DOMString name, OperandDescriptor desc);
+  // Create an operand for a model constant from a WebGL buffer.
+  MLOperand constant(MLOperandDescriptor desc, WebGLBuffer glBuffer, 
+                     optional long offset = 0, optional long size);
 
-  // Create an operand that represents a model constant.
-  Operand constant(OperandDescriptor desc, ArrayBufferView value);
+  // Create an operand for a model constant from a WebGPU buffer.
+  MLOperand constant(MLOperandDescriptor desc, GPUBuffer gpuBuffer, 
+                     optional long offset = 0, optional long size);
```

Comment:
Would it make sense to introduce the `MLBuffer` that supports `ArrayBuffer`, `WebGLBuffer` and `GPUBuffer` by WebIDL [union type](https://heycam.github.io/webidl/#idl-union)? The `offset` and `size` probably need to be `unsigned long long` as `GPUBuffer` uses it.

e.g.
```js
dictionary MLBuffer {
  required (ArrayBuffer or WebGLBuffer or GPUBuffer) buffer;
  unsigned long long offset = 0;
  unsigned long long size;
};

MLOperand constant(MLOperandDescriptor desc, MLBuffer buffer);
```

The `compute` method can also use `MLBuffer`.

Reply:
Good idea. I'll make that change.

Reply:
Looks good with an open, commented inline. Thanks!